diff --git a/app/opmip-lma/Jamfile b/app/opmip-lma/Jamfile
index 7a2cbd9..f49694f 100644
--- a/app/opmip-lma/Jamfile
+++ b/app/opmip-lma/Jamfile
@@ -19,8 +19,6 @@ exe opmip-lma
 	: main.cpp
 	  options.cpp
 	  ../../lib/opmip
-	  /boost//thread
-	  /boost//program_options
 	;
 
 install install
#diff --git a/app/opmip-mag/Jamfile b/app/opmip-mag/Jamfile
#index 51f82f4..88e0f84 100644
#--- a/app/opmip-mag/Jamfile
#+++ b/app/opmip-mag/Jamfile
#@@ -38,8 +38,6 @@ exe opmip-mag
# 	  drivers/madwifi_driver_impl.cpp
# 	  drivers/dummy.cpp
# 	  ../../lib/opmip
#-	  /boost//thread
#-	  /boost//program_options
# 	  dl
# 	  ieee802_11
# 	;
diff --git a/lib/opmip/Jamfile b/lib/opmip/Jamfile
index ec2d35c..56237af 100644
--- a/lib/opmip/Jamfile
+++ b/lib/opmip/Jamfile
@@ -20,7 +20,6 @@ opmip.runtime-lib librt : rt ;
 
 lib conf
 	: conf.cpp
-	  /boost//headers
 	: <c++-template-depth>1024
 	;
 
@@ -48,8 +47,6 @@ lib opmip
 	  pmip/addrconf_server.cpp
 	  sys/ip6_tunnel_service.cpp
 	  sys/route_table.cpp
-	  /boost//headers
-	  /boost//system
 	  pthread
 	  librt
 	;

